_source/_posts/2022-06-01-jhipster-digital-ocean.md

```diff
@@ -0,0 +1,442 @@
+
+For this test, start from the `reactive-jhipster` example in the `java-microservices-examples` repository on [GitHub](https://github.com/oktadev/java-microservices-examples). The example is a JHipster reactive microservices architecture with Spring Cloud Gateway and Spring WebFlux, Vue as the client framework, and Gradle as the build tool. You can read about how it was built in [Reactive Java Microservices with Spring Boot and JHipster](/blog/2021/01/20/reactive-java-microservices).
+
+```bash
+git clone https://github.com/oktadev/java-microservices-examples.git
+cd java-microservices-examples/reactive-jhipster
+```
+If you inspect the project folder, you will find sub-folders for the `gateway` service, which will act as the front-end application, and a gateway to the `store` and `blog` microservices, which also have their subfolders. A `docker-compose` sub-folder contains the service definitions for running the application containers.
+
+The next step is to generate the Kubernetes deployment descriptors. In the project root folder, create a `kubernetes` directory and run the `k8s` JHipster sub-generator:
+
+```bash
+mkdir kubernetes
+cd kubernetes
+jhipster k8s
+```
+
+Choose the following options when prompted:
+
+- Type of application: **Microservice application**
+- Root directory: **../**
+- Which applications? (select all)
+- Set up monitoring? **No**
+- Which applications with clustered databases? select **store**
+- Admin password for JHipster Registry: (generate one)
+- Kubernetes namespace: **demo**
+- Docker repository name: (your docker hub username)
+- Command to push Docker image: `docker push`
+- Enable Istio? **No**
+- Kubernetes service type? **LoadBalancer**
+- Use dynamic storage provisioning? **Yes**
+- Use a specific storage class? (leave empty)
+
+**NOTE**: You can leave the Docker repository name blank for running Kubernetes locally, but a repository will be required for the cloud deployment, so go ahead and create a [DockerHub](https://hub.docker.com/) personal account, and the image pull configuration will be ready for both local and cloud deployments.
+
+Build the `gateway`, `store` and `blog` services container images with Jib. As Jib will push the images to DockerHub, you first need to do a `docker login`.
+
+```bash
+docker login
+```
+If you have two-factor authentication enabled, you must generate a token and use it as the password for the login. In the Docker web, go to the user menu and choose **Account Settings**. Then in the left menu choose **Security** and **New Access Token**.
+
+For example, for building the `gateway` service image:
+
+```bash
+cd ../gateway
+./gradlew bootJar -Pprod jib -Djib.to.image=<docker-repo-name>/gateway
+```
+**IMPORTANT NOTE**: Unfortunately the application example does not build with Java 17 at the moment of writing this post.
```

Comment:
Should we remind them here what Java build *is* supported?

Reply:
Added a remider.